### ISSUE TITLE
Content-Type getting set incorrectly in CurlHandler

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3124]
This test passes on Windows but fails on Unix:
```C#
[Theory, MemberData("PostServers")]
public async Task PostAsync_CallMethod_StreamContent(Uri remoteServer)
{
    using (var handler = new HttpClientHandler())
    using (var client = new HttpClient(handler))
    {
        byte[] data = new byte[1234];
        new Random(42).NextBytes(data);
        HttpContent content = new StreamContent(new MemoryStream(data));
        HttpResponseMessage response = await client.PostAsync(remoteServer, content);
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        string responseContent = await response.Content.ReadAsStringAsync();
        Assert.Contains(Convert.ToBase64String(data), responseContent);
    }
}
```
It looks like the Content-Type header is being set inappropriately.

**Comment by davidsh** [comment]
Not specific to this issue but I thought i would mention this...

```c#
using (var handler = new HttpClientHandler())
using (var client = new HttpClient(handler))
```

It isn't necessary to have a `using` for the handler.  Disposing the `HttpClient` object will automatically dispose the `HttpClientHandler` object.

**Comment by stephentoub** [comment]
Fixed by #3131
